kernel/api/src/main/java/org/sakaiproject/tasks/api/TaskService.java

```diff
@@ -23,13 +23,18 @@
 
     UserTask createSingleUserTask(UserTaskAdapterBean transfer);
     UserTask saveUserTask(UserTaskAdapterBean transfer);
+    UserTask saveUserTask(Task task, String userId, Integer priority, String notes);
     void removeUserTask(Long userTaskId);
     Task createTask(Task task, Set<String> users, Integer priority);
     Task saveTask(Task task);
     Optional<Task> getTask(String reference);
     List<UserTaskAdapterBean> getAllTasksForCurrentUser();
+    List<UserTaskAdapterBean> getAllTasksForCurrentUserOnSite(String siteId);
     List<UserTask> getCurrentUserTasks(String userId);
     void removeTask(Task task);
     void removeTaskByReference(String reference);
     void completeUserTaskByReference(String reference, List<String> userIds);
+    void assignTask(Task task, String type, String objectId);
+    void assignTask(Task task, String type, List<String> objectIds);
```

Comment:
I'd make these one method and just check the list. It seems a bit overkill to have two methods.

Reply:
Overloading methods with different types in a service makes easier to work on the controller side. It's different to assign a task to a single object or to do the same to a collection of them.